models/enterprise_sonic/snmp/sonic_snmp.yml

```diff
@@ -1,53 +1,265 @@
+                required: true
+              port:
+                description:
+                  - UDP listening port, range 1024-65535
+                type: int
+              interface:
+                description:
+                  - Name of interface
+                  - interface and VRF are mutually exclusive
+                type: str
+              vrf:
+                description:
+                  - Name of VRF
+                  - VRF and interface are mutually exclusive
+                type: str
+          community:
+            description:
+              - List of SNMP community configuration
+            type: list
+            elements: dict
+            suboptions:
+              group:
+                description:
+                  - Name of community group
+                type: str
+              name:
+                description:
+                  - Name of community
+                type: str
+                required: true
+          engine:
+            description:
+              - SNMP engine ID
+            type: str
+          user:
+            description:
+              - List of SNMP user configuration
+            type: list
+            elements: dict
+            suboption:
+              group:
+                description:
+                  - Name of user group
+                type: str
+              name:
+                description:
+                  - Name of user
+                type: str
+                required: true
+              auth:
+                description:
+                  - User authentication configuration
+                type: dict
+                suboptions:
+                  md5_key: 
+                    description: 
+                      - MD5 authentication key
+                    type: str
+                  sha_key:
+                    description:
+                      - SHA authentication key
+                    type: str
+              priv:
+                description:
+                  - Privacy configuration
+                type: dict
+                suboptions:
+                  des_key:
+                    description:
+                      - DES ecryption
+                    type: str
+                  aes_key:
+                    description:
+                      - AES encryption
+                    type: str
+              encrypted:
+                description:
+                  - Enable/disable encryption
+                type: bool       
+          view:
+            description:
+             - List of SNMP view configuration
+            type: list
+            elements: dict
+            suboptions:
+              name:
+                description:
+                  - Name of the view
+                type: str
+                required: true
+              included:
+                description:
+                  - List of OID trees included in the view
+                type: list
+                elements: str
```

Comment:
It looks like there is a format requirement for the "included" and "excluded" strings. Please check on it and put some description of the requirement in this section.